lib/indexer.rb

```diff
@@ -0,0 +1,248 @@
+      end
+
+      msgs << "Error count (items + coll record w any error; may have indexed on retry if it was a timeout): #{metrics.error_count}"
+#      msgs << "Retry count: #{@retries}"  # currently useless due to bug in harvestdor-indexer 0.0.12
+      msgs << "Total records processed: #{metrics.total}"
+      msgs
+    end
+  end
+
+  # log details about the results of indexing
+  def log_results
+    record_count_msgs.each { |msg|
+      logger.info msg
+    }
+    logger.info("Avg solr commit time per object (successful): #{(@total_time_to_solr/metrics.success_count).round(2)} seconds") unless metrics.success_count == 0
+    logger.info("Avg solr commit time per object (all): #{(@total_time_to_solr/metrics.total).round(2)} seconds") unless metrics.total == 0
+    logger.info("Avg parse time per object (successful): #{(@total_time_to_parse/metrics.success_count).round(2)} seconds") unless metrics.success_count == 0
+    logger.info("Avg parse time per object (all): #{(@total_time_to_parse/metrics.total).round(2)} seconds") unless metrics.total == 0
+    logger.info("Avg complete index time per object (successful): #{(@total_time/metrics.success_count).round(2)} seconds") unless metrics.success_count == 0
+    logger.info("Avg complete index time per object (all): #{(@total_time/metrics.total).round(2)} seconds") unless metrics.total == 0
+  end
+
+  def email_report_body
+    body = ''
+    body += "\n" + record_count_msgs.join("\n") + "\n"
+
+    if @druids_failed_to_ix.size > 0
+      body += "\n"
+      body += "records that may have failed to index (merged recs as druids, not ckeys): \n"
+      body += @druids_failed_to_ix.join("\n") + "\n"
+    end
+
+    body += "\n"
+    body += "full log is at gdor_indexer/shared/#{config.harvestdor.log_dir}/#{config.harvestdor.log_name} on #{Socket.gethostname}"
+    body += "\n"
+
+    body += @validation_messages.join("\n") + "\n"
+  end
+
+  # email the results of indexing if we are on one of the harvestdor boxes
+  def email_results
+    if config.notification
+      to_email = config.notification
+
+      opts = {}
+      opts[:subject] = "#{config.harvestdor.log_name.chomp('.log')} into Solr server #{config[:solr][:url]} is finished"
+      opts[:body] = email_report_body
+      begin
+        send_email(to_email, opts)
+      rescue => e
+        logger.error('Failed to send email notification!')
+        logger.error(e)
+      end
+    end
+  end
+
+  def send_email(to, opts = {})
+    opts[:server]     ||= 'localhost'
+    opts[:from]       ||= 'gryphondor@stanford.edu'
+    opts[:from_alias] ||= 'gryphondor'
+    opts[:subject]    ||= 'default subject'
+    opts[:body]       ||= 'default message body'
+    mail = Mail.new do
+      from    opts[:from]
+      to      to
+      subject opts[:subject]
+      body    opts[:body]
+    end
+    mail.deliver!
+  end
+
+  def elapsed_time(start_time, units = :seconds)
+    elapsed_seconds = Time.now - start_time
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.